docs/spec/connection-handshake/L1_2.md

```diff
@@ -0,0 +1,264 @@
+
+
+- **[CH/Seq/2/Termination]** Eventually, the two processes should both finish the handshake, establishing a connection.
+
+Requirement **[CH/Seq/1/Agreement]** represents the basic safety guarantee, while **[CH/Seq/2/Termination]** is the basic liveness guarantee we would like to have.
+
+The diagram below sketches the high-level problem statement.
+
+```
+                        IBC Connection
+     +-----------+    Handshake Protocol    +-----------+
+     |           |<-------- step 1 -------->|           |
+     | Initiator |    (execute handshake)   | Responder |
+     |           |                          |           |
+     |          _| (connection established) |_          |
+     |         |c|<------------------------>|c|         |
+     |          ▔|                          |▔          |
+     +-----------+                          +-----------+
+```
+
+
+## L2. Protocol specification / protocol view
+
+In this section we attempt to make the problem specification more concrete.
+We define what the processes are, their fault-model, as well as communication and synchrony assumptions.
+
+### 2.1 System model specification
+
+#### System model
+
+A **chain** is a state machine.
+Each chain may be replicated, i.e., consisting of a replicated state machine (RSM), or it may be a standalone process.
+In practice, there are certain requirements which each chain must satisfy (e.g., see [[ICS 024](#references)]).
+At this level of specification, however, we can assume each chain is a single **process**.
+
+The connection handshake protocol is asymmetric, meaning that the two chains that interact play different roles within the protocol.
+We take advantage of this asymmetry and explicitly label one of the chains with `I` (the initiating, or source, process), and the other with `R` (the responder, or destination, process).
+
+On top of each chain, one or multiple **applications** may be running (e.g., payment system or token exchanges, with their associated smart contracts).
+In the context of the connection handshake problem, it is an application that invokes the connection handshake protocol, and provides the necessary input for executing this protocol (we give more details on the invocation & input in section [2.2 Protocol](#22-protocol) below).
+
+At a high-level, the handshake protocol consists of processes `I` and `R` exchanging a few messages called **datagrams**.
+Each datagram has a well-defined syntax which we describe later.
+__[FIX: describe this and put ref to [2.2 Protocol](#22-protocol).]__
+
+We define a **connection** as a bidirectional or unidirectional path of communication that applications on the initiator and responder chains use to exchange **packets**.
+Packets are payloads from the point of view of a connection; they contain arbitrary application-level data, and their content is opaque to a connection; suffice to say that applications generate and consume packets.
```

Comment:
I am not sure about the correctness of these statements, what does it mean exactly "content is opaque to a connection"? or "packets are payloads from the point of view of a connection"? Where are the channels for example? The packets carry port and channel identifiers.